### PR TITLE
Add mode parameter to choose between quickpicklist and txt

### PR DESCRIPTION
**Pull request by matedabis**
**Add mode parameter to choose between quickpicklist and txt when running multiple files**

There are 3 modes available:

1. Running a single js file
2. Running multiple files with the quickpicklist selection (original solution)
3. Running multiple files with the txt list selection (new solution added in #14 )

___

1. Running a single js file:

Needs a similar launch.json file, assuming the file we would like to run is called "t.js" which is in our working directory.
```
{
  "version": "0.2.0",
  "configurations": [

    
    {
      "name": "Escargot.js: Launch",
      "type": "escargot",
      "request": "launch",
      "program": "/home/dabis/work/jerryscript/escargot/escargot",
      "address": "localhost",
      "localRoot": "${workspaceRoot}",
      "debugLog": 0,
      "args": [
        "--debugger-wait-source",
        "--start-debug-server",
        "./t.js"
    ]
    }
  ]
}
```

2. Running multiple files with the quickpicklist selection

Set the added "wait_for_source_mode" parameter to "quickpicklist"

```
{
  "version": "0.2.0",
  "configurations": [

    
    {
      "name": "Escargot.js: Launch",
      "type": "escargot",
      "request": "launch",
      "program": "/home/dabis/work/jerryscript/escargot/escargot",
      "address": "localhost",
      "localRoot": "${workspaceRoot}",
      "debugLog": 0,
      "wait_for_source_mode": "quickpicklist",
      "args": [
          "--start-debug-server",
          "--debugger-wait-source",
      ]
    }
  ]
}

```
3. Running multiple files with the txt list selection

Set the "wait_for_source_mode" parameter to the txt filename
```

{
  "version": "0.2.0",
  "configurations": [

    
    {
      "name": "Escargot.js: Launch",
      "type": "escargot",
      "request": "launch",
      "program": "/home/dabis/work/jerryscript/escargot/escargot",
      "address": "localhost",
      "localRoot": "${workspaceRoot}",
      "debugLog": 0,
      "wait_for_source_mode": "asd.txt",
      "args": [
          "--start-debug-server",
          "--debugger-wait-source",
      ]
    }
  ]
}
```

_Note: Set path to escargot as yours._

### CODE REVIEW

**Comment by matedabis** [comment]
Running 1 file also works the following way:

Start escargot in cmd
`../escargot/escargot --start-debug-server ../escargot-test/t.js `

Set launch.json  to:
`{
  "version": "0.2.0",
  "configurations": [

    
    {
      "name": "Escargot.js: Attach",
      "type": "escargot",
      "request": "attach",
      "program": "/home/dabis/work/jerryscript/escargot/escargot",
      "address": "localhost",
      "localRoot": "${workspaceRoot}",
      "debugLog": 0,

    }
  ]
}`

File: src/extension.ts [hunk 1]
```diff
@@ -74,54 +65,86 @@ const getProgramSource = (path: string[]): string[] => {
...
-            name: pathArray.shift(),
-            source: sources.shift(),
-            isLast: sources.length === 0
-          };
         }
+      }
 
-        vscode.debug.activeDebugSession.customRequest('sendSource', { program });
-        return true;
+      else if(e.body == "quickpicklist") {
```
Comment (zherczeg) [inline src/extension.ts:115]: Perhaps we should check this case first, and use the filename in all other cases. Currently it is possible that nothing happens.

File: src/extension.ts [hunk 2]
```diff
@@ -74,54 +65,86 @@ const getProgramSource = (path: string[]): string[] => {
...
+      source: sources.shift(),
+      isLast: sources.length === 0
+    };
+  }
+  let counter = 0;
+  while(!vscode.debug.activeDebugSession) {
+    await new Promise(f => setTimeout(f, 100));
+    counter++;
+    if (counter > 100) {
+      return undefined;
```
Comment (zherczeg) [inline src/extension.ts:143]: You can do this with less code:
```
    counter++;
    if (counter > 100) {
```
to
```
    if (++counter > 100) {
```

The name should be "retries" not "counter".

File: src/extension.ts [hunk 3]
```diff
@@ -74,54 +65,86 @@ const getProgramSource = (path: string[]): string[] => {
...
+  let program = {};
+  if (sources.length) {
+    program = {
+      name: pathArray.shift(),
+      source: sources.shift(),
+      isLast: sources.length === 0
+    };
+  }
+  let counter = 0;
+  while(!vscode.debug.activeDebugSession) {
```
Comment (zherczeg) [inline src/extension.ts:139]: Space after while. You could add some newlines to this code block to make it more readable.

**Comment by matedabis** [comment]
Updated according to the reviews

File: src/extension.ts [hunk 4]
```diff
@@ -74,54 +65,89 @@ const getProgramSource = (path: string[]): string[] => {
...
+      source: sources.shift(),
+      isLast: sources.length === 0
+    };
   }
+
+  let retries = 0;
+
+  while (!vscode.debug.activeDebugSession) {
+
+    await new Promise(f => setTimeout(f, 100));
```
Comment (zherczeg) [inline src/extension.ts:143]: I would put the newline after the await, not before.

File: src/extension.ts [hunk 5]
```diff
@@ -74,54 +65,89 @@ const getProgramSource = (path: string[]): string[] => {
...
-                vscode.window.showInformationMessage(path.join(folder_path,data[i]));
-              }
+  switch (eventType) {
+    case 'readSources': {
+      pathArray = [];
+
+      if(e.body == "quickpicklist") {
+        await getProgramName().then(path => path);
+      }
+
```
Comment (zherczeg) [inline src/extension.ts:103]: Remove this newline

**Review by zherczeg — APPROVED** [review]
LGTM with minor changes

**Comment by matedabis** [comment]
Updated.

File: src/extension.ts [hunk 6]
```diff
@@ -74,54 +65,89 @@ const getProgramSource = (path: string[]): string[] => {
...
+      name: pathArray.shift(),
+      source: sources.shift(),
+      isLast: sources.length === 0
+    };
   }
+
+  let retries = 0;
+
+  while (!vscode.debug.activeDebugSession) {
+
```
Comment (zherczeg) [inline src/extension.ts:142]: Remove this newline

**Pull request merged**